libdnf/system/state.cpp

```diff
@@ -262,6 +275,8 @@ GroupState State::get_group_state(const std::string & id) {
 
 void State::set_group_state(const std::string & id, const GroupState & group_state) {
     group_states[id] = group_state;
+    // invalidate the package groups cache
+    package_groups_cache.reset();
```

Comment:
You also need this in `remove_group_state()` right below. Maybe it should also be added to `State::load()` for completeness, though it may not ever get used that way...

Reply:
oh, righ. good catch. I'll fix it.